src/hooks/use-presentation.test.js

```diff
@@ -1,61 +1,101 @@
-import React from 'react';
-import Enzyme, { mount } from 'enzyme';
+import 'regenerator-runtime/runtime';
+import Enzyme from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
-
+import { renderHook, act } from '@testing-library/react-hooks';
 import usePresentation from './use-presentation';
 
 Enzyme.configure({ adapter: new Adapter() });
 
 describe('usePresentation', () => {
-  it('when starting connection it calls start method on PresentationRequest and connection is set to true', () => {
-    const startMock = jest.fn().mockResolvedValue({});
-    class PresentationRequest {
-      constructor() {}
-      start = startMock;
-    }
-    global.PresentationRequest = PresentationRequest;
-    const TestComponent = () => {
-      const { startConnection } = usePresentation();
-      return (
-        <button
-          data-testid="startConnection"
-          onClick={() => startConnection()}
-        ></button>
-      );
-    };
-    const component = mount(<TestComponent />);
-    component.find('[data-testid="startConnection"]').simulate('click');
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.clearAllTimers();
+  });
+
+  it('when starting connection it calls start method on PresentationRequest.', async () => {
+    const startMock = jest.fn().mockResolvedValue({ terminate: () => {} });
+    global.PresentationRequest = jest
+      .fn()
+      .mockImplementationOnce(() => ({ start: startMock }));
+    const {
+      result: { current },
```

Comment:
@upatel32 So this is a strange case on where we cannot destructure current from result on line 45 because the getter recalculates the value by that time in the async block.